library/windows/BUILD.gn

```diff
@@ -0,0 +1,82 @@
+# Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import("//build/packaging.gni")
+import("//library/engine.gni")
+import("//library/windows/glfw.gni")
+import("//library/windows/jsoncpp.gni")
+
+action("fetch_glfw") {
+  script = "//tools/dart_tools/bin/fetch_glfw.dart"
+  outputs = glfw_files
+  args = [
+    rebase_path(glfw_download_dir, root_build_dir),
+  ]
+  public_configs = [
+    "//build/win/config:glfw3",
+  ]
+}
+
+config("relative_glfw_dependencies") {
+  include_dirs = [
+    glfw_include_dir,
+  ]
+  lib_dirs = [
+    glfw_download_dir,
+  ]
+}
+
+action("build_jsoncpp") {
```

Comment:
In case this lasts longer than I hope, let's document the limitations here. Something like:
```
# TODO: Replace this with actual GN build rules for jsoncpp.
# This target is a placeholder, and will not actually rebuild jsoncpp if any of its
# source changes.
```